src/Processors/Formats/Impl/ArrowBufferedStreams.h

```diff
@@ -62,6 +62,9 @@ class RandomAccessFileFromSeekableReadBuffer : public arrow::io::RandomAccessFil
 
     arrow::Result<std::shared_ptr<arrow::Buffer>> Read(int64_t nbytes) override;
 
+    /// Override async reading to avoid using internal arrow thread pool.
+    arrow::Future<std::shared_ptr<arrow::Buffer>> ReadAsync(const arrow::io::IOContext&, int64_t position, int64_t nbytes) override;
```

Comment:
I guess it would be good to mention that this call is actually sync and not async.